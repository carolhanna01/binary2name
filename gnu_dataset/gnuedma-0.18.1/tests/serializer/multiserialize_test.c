/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdio.h>
#include <edma.h>

int main (int argc, char *argv[])
{
  OBJID		id, id1;
  EChar         a[80];
  
  EDMAInit();
  id1 = edma_new_obj ("MULTI_SERIALIZER");
  id = edma_new_obj ("ID");
  
  edma_met3 (id, "SetDB", "test.db");
  /* Add one superobject and one subobject*/
  edma_wprop3 (id, "AP_TEST@INNER<Id", "SUBOBJECT");
  //edma_wprop3 (id, "PERSON@INNER<name", "John Doe");
  edma_wprop3 (id, "AP_TEST@SUPER>Id", "SUPEROBJECT");


  //edma_wprop3 (id, "INNER<Id", "XXXXXX");
  edma_rprop3 (id, "SUPER>Id", a);
  printf ("SUPER: %s\n", a);
  edma_rprop3 (id, "INNER<Id", a);
  printf ("INNER: %s\n", a);

  edma_obj_report (id);
  edma_show_subobjects_up (id, "--", 0);
  edma_show_subobjects_down (id, "--", 0);
  /* Marshall compound object*/
  edma_met3 (id1, "marshall", id);
  edma_met3 (id1, "save", "multidata.dat");
  edma_free_obj (id);
  edma_free_obj (id1); 
  printf ("------------------------------------------------------\n");
  EDMAEnd();
}
