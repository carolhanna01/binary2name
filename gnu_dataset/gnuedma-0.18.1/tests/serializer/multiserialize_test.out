
GNU/EDMA 0.17.1 build [Feb  6 2010][Beta][0.17.1] starting up...
EDMA Startup process... OK
SUPER: SUPEROBJECT
INNER: SUBOBJECT
==OBJECT REPORT==================================================
Object (1) of class (ID)
Top Level Object. Object haven't father
No PseudoFather information
Object has 1 superobjects
    Id : (SUPER) Class : (AP_TEST) Class attached Object: (AP_TEST)
Object has 1 subobjects
    Id : (INNER) Class : (AP_TEST) Class attached Object: (AP_TEST)
Virtual Methods in object
==END OBJECT REPORT==========================================
-- ==> ID [1]
  SUPER ==> AP_TEST [3]
  INNER ==> AP_TEST [2]
-- ==> ID [1]
------------------------------------------------------
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
